test/extensions/common/wasm/wasm_vm_test.cc

```diff
@@ -40,6 +41,25 @@ proxy_wasm::RegisterNullVmPluginFactory register_test_null_vm_plugin("test_null_
   return plugin;
 });
 
+class NoneVmTest : public testing::Test {
+  void SetUp() override {
+    original_factories_ = Registry::FactoryRegistry<WasmRuntimeFactory>::factories();
+    Registry::FactoryRegistry<WasmRuntimeFactory>::factories().clear();
```

Comment:
Clear when setup not teardown?

Reply:
Yes. To simulate no VM case, all the factories are cleared here. It is recovered when "TearDown".